modules/core/misc/python/package/mat_wrapper/__init__.py

```diff
@@ -0,0 +1,33 @@
+__all__ = []
+
+import sys
+import numpy as np
+import cv2 as cv
+
+# NumPy documentation: https://numpy.org/doc/stable/user/basics.subclassing.html
+
+class Mat(np.ndarray):
+    '''
+    cv.Mat wrapper for numpy array.
+
+    Stores extra metadata information how to interpret and process of numpy array for underlying C++ code.
+    '''
+
+    def __new__(cls, arr, **kwargs):
+        obj = arr.view(Mat)
+        return obj
+
+    def __init__(self, arr, **kwargs):
```

Comment:
Why `__init__` is required?
I think `Mat` is [RealisticInfoArray](https://numpy.org/doc/stable/user/basics.subclassing.html#slightly-more-realistic-example-attribute-added-to-existing-array) from the user guide except that `Mat` should wrap only numeric types. For Python objects and strings it should raise `AssertionError` 

And it is better to explicitly add `wrap_channels` argument to `__new__` with the documentation string.